pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -399,9 +402,118 @@ func (r *rbac) createRoleBinding(ctx context.Context, sa *corev1.ServiceAccount)
+	viewCRB, err := rbacClient.ClusterRoleBindings().Get(ctx, "clusterinterceptors-clusterrolebinding", metav1.GetOptions{})
+
+	if err == nil {
+		logger.Infof("found clusterrolebinding %s", viewCRB.Name)
+		return r.updateClusterRoleBinding(ctx, viewCRB, sa)
+	}
+
+	if errors.IsNotFound(err) {
+		return r.createClusterRoleBinding(ctx, sa)
 	}
+
 	return err
 }
+
+func (r *rbac) updateClusterRoleBinding(ctx context.Context, rb *rbacv1.ClusterRoleBinding, sa *corev1.ServiceAccount) error {
+	logger := logging.FromContext(ctx)
+
+	subject := rbacv1.Subject{Kind: rbacv1.ServiceAccountKind, Name: sa.Name, Namespace: sa.Namespace}
+
+	hasSubject := hasSubject(rb.Subjects, subject)
+	if !hasSubject {
+		rb.Subjects = append(rb.Subjects, subject)
+	}
+
+	ownerRef := rb.GetOwnerReferences()
+	if len(ownerRef) == 0 {
+		rb.SetOwnerReferences([]metav1.OwnerReference{r.ownerRef})
+	}
+
+	if hasSubject && (len(ownerRef) != 0) {
+		logger.Info("clusterrolebinding is up to date", "action", "none")
+		return nil
+	}
+
+	logger.Info("update existing clusterrolebinding clusterinterceptors-clusterrolebinding")
+	rbacClient := r.kubeClientSet.RbacV1()
+
+	if _, err := rbacClient.ClusterRoleBindings().Update(ctx, rb, metav1.UpdateOptions{}); err != nil {
+		logger.Error(err, "failed to update clusterinterceptors-clusterrolebinding crb")
+		return err
+	}
+	logger.Info("successfully updated clusterinterceptors-clusterrolebinding")
+	return nil
+}
+
+func (r *rbac) createClusterRoleBinding(ctx context.Context, sa *corev1.ServiceAccount) error {
+	logger := logging.FromContext(ctx)
+
+	logger.Info("create new clusterrolebinding clusterinterceptors-clusterrolebinding")
+	rbacClient := r.kubeClientSet.RbacV1()
+
+	logger.Info("finding clusterrole clusterinterceptors-clusterrole")
+	if _, err := rbacClient.ClusterRoles().Get(ctx, "clusterinterceptors-clusterrole", metav1.GetOptions{}); err != nil {
+		logger.Error(err, " getting clusterRole 'clusterinterceptors-clusterrole' failed")
+		return err
+	}
+
+	crb := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "clusterinterceptors-clusterrolebinding",
+			OwnerReferences: []metav1.OwnerReference{r.ownerRef},
+		},
+		RoleRef:  rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: "clusterinterceptors-clusterrole"},
+		Subjects: []rbacv1.Subject{{Kind: rbacv1.ServiceAccountKind, Name: sa.Name, Namespace: sa.Namespace}},
+	}
+
+	if _, err := rbacClient.ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{}); err != nil {
+		logger.Error(err, " creation of 'clusterinterceptors-clusterrolebinding' failed")
+		return err
+	}
+	return nil
+}
+
+func (r *rbac) createClusterRole(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+
+	logger.Info("create new clusterrole clusterinterceptors-clusterrole")
+	rbacClient := r.kubeClientSet.RbacV1()
+
+	cr := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "clusterinterceptors-clusterrole",
```

Comment:
let us add openshift-pipelines- prefix to this name and
we can skip the clusterrole suffix, as it is redundant.
In addition, let us keep this name in a constant.